### ISSUE TITLE
Deprecate twisted.web.trp

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2030]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2030|
|Type|defect|
|Created|2006-08-29 07:13:59Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-trp-2030|

"""I contain ResourceUnpickler, which will unpickle any python object"""

What does this mean?

<details><summary>Searchable metadata</summary>

```
trac-id__2030 2030
type__defect defect
reporter__jml jml
priority__normal normal
milestone__ 
branch__branches_deprecate_trp_2030 branches/deprecate-trp-2030
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1156835639000000 1156835639000000
changetime__1233347825000000 1233347825000000
version__None None
owner__ 
cc__spiv cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|spiv commented|
|-|-|

I'm guessing that if we simply deleted this module no-one would notice.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

I tend to agree, but let's deprecated it first anyway.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_2):
> I tend to agree, but let's deprecated it first anyway.

Changing the ticket title in that case.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [26016]) Branching to 'deprecate-trp-2030'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

I still don't know how to test module deprecations. :(  Although I am starting to have ideas!

Anyway, some code in a branch, please review.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_6"></a>|@mwhudson set owner to @exarkun|
|-|-|

twisted.web.tap imports trp, is that deprecated too?  Guess: no.

It should be easy enough to work around this, I think.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun removed owner|
|-|-|

Not incredibly hard, anyway.  I made `twisted.web.tap` not emit the deprecation warning (r26060) just for being imported; one is imported if you actually serve a trp, though.  I looked around to see if there was anything else related to trp and found File.createPickleChild, which I deprecated in r26061.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_8"></a>|@mwhudson set owner to @exarkun|
|-|-|

Well, I'm happy with all the code I read last time now... and I guess the new test_path increases test coverage, but I don't think that it's a particularly good test.  It's docstring is very vague, and comments like "# Various sorts of children should be processed in variety of ways." are pretty close to useless.  I would prefer to see a bunch of single condition tests (http://xunitpatterns.com/Principles%20of%20Test%20Automation.html#Verify%20One%20Condition%20per%20Test) for each processor, using test methods to minimize the duplication.

But never mind, merge the lot anyway if you can't be bothered to do that.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [26125]) split test_path up into a bunch of tiny pieces

refs [#2030](https://github.com/twisted/twisted/issues/2030)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun removed owner|
|-|-|

I think I improved the situation in r26125.  How's that?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set owner to @exarkun|
|-|-|

 * Some flakes:
```
twisted/web/test/test_tap.py:24: 'ResourceScript' imported but unused
twisted/web/test/test_tap.py:25: 'ResourceUnpickler' imported but unused
```
 * There is a dumb failure in `test_trpProcessor` (be careful, it changes once you fix the imports).

Please merge after those fixes.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [26134]) Fix unused imports and a stupid bug in test_trpProcessor

refs [#2030](https://github.com/twisted/twisted/issues/2030)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

(In [26135]) Handle the possibility of a module-level deprecation

refs [#2030](https://github.com/twisted/twisted/issues/2030)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun set status to `closed`|
|-|-|

(In [26136]) Merge deprecate-trp-2030 

Author: exarkun
Reviewer: mwhudson, therve
Fixes: [#2030](https://github.com/twisted/twisted/issues/2030)

Deprecate `twisted.web.trp`, a poorly conceived resource persistence
system for Twisted Web.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
